File: PUGD-frontend/graphql/queries/acquisition/suggestion.js
import gql from "graphql-tag";

const GetSuggestions = gql`
  query($flag: String!) {
    getsuggestions(flag: $flag) {
      _id
      isbn
      title
      author
      quantity
      price
      datepublication
      comments
      source
    }
  }
`;

const GetSuggestionsReader = gql`
  query($source: String!) {
    getsuggestions(source: $source) {
      _id
      isbn
      title
      author
      quantity
      price
      datepublication
      comments
      source
    }
  }
`;

module.exports = {
  GetSuggestions,
  GetSuggestionsReader,
};

File: PUGD-api/graphql/types/AcquisitionTypes/Suggestion.go
package AcquisitionTypes

import "github.com/graphql-go/graphql"

var Suggestion = graphql.NewObject(graphql.ObjectConfig{
	Name: "Suggestion",
	Fields: graphql.Fields{
		"_id": &graphql.Field{
			Type: graphql.ID,
		},
		"isbn": &graphql.Field{
			Type: graphql.String,
		},
		"title": &graphql.Field{
			Type: graphql.String,
		},
		"author": &graphql.Field{
			Type: graphql.String,
		},
		"quantity": &graphql.Field{
			Type: graphql.Int,
		},
		"price": &graphql.Field{
			Type: graphql.Float,
		},
		"datepublication": &graphql.Field{
			Type: graphql.String,
		},
		"comments": &graphql.Field{
			Type: graphql.String,
		},
		"source": &graphql.Field{
			Type: graphql.String,
		},
		"flag": &graphql.Field{
			Type: graphql.String,
		},
	},
})
